src/pages/login/login.ts

```diff
@@ -23,38 +27,52 @@ export class LoginPage extends BasePageComponent {
   deviceTypeError: DeviceError;
   hasUserLoggedOut: boolean = false;
   hasDeviceTypeError: boolean = false;
+  unauthenticatedMode: boolean = false;
 
   constructor(
     public navCtrl: NavController,
     public navParams: NavParams,
     public platform: Platform,
     public splashScreen: SplashScreen,
     private store$: Store<StoreModel>,
+    public networkStateProvider: NetworkStateProvider,
     public authenticationProvider: AuthenticationProvider,
     public appConfigProvider: AppConfigProvider,
     public analytics: AnalyticsProvider,
     public deviceProvider: DeviceProvider,
+    private secureStorage: SecureStorage,
+    public dataStore: DataStoreProvider,
+
   ) {
     super(platform, navCtrl, authenticationProvider, false);
 
     // Check to see if redirect to page was from a logout
     this.hasUserLoggedOut = navParams.get('hasLoggedOut');
+    this.networkStateProvider.initialiseNetworkState();
 
     // Trigger Authentication if this isn't a logout and is an ios device
     if (!this.hasUserLoggedOut && this.isIos()) {
       this.login();
     }
-
     if (!this.isIos()) {
+      this.appConfigProvider.initialiseAppConfig();
       this.navController.setRoot('JournalPage');
       this.splashScreen.hide();
     }
   }
 
-  login = (): Promise<any> =>
+  login = async (): Promise<any> => {
+    await this.platform.ready();
```

Comment:
Is there a way to chain this in with the other platform.ready. It seems strange to be waiting on platform.ready twice in the same function

Reply:
Yes, shouldn't platform already be ready as it was awaited at the beginning of the function?

Reply:
Have promisified the initalise app config and chained from that.